### PR TITLE
FC-1100  Permissions System  

### PR DESCRIPTION
**Pull request by eplazaso**

### CODE REVIEW

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md [hunk 1]
```diff
@@ -8,7 +8,199 @@ permalink: liveperson-functions-permission-system.html
...
 ---
 
-# Permission System
+There are 3 user permissions pertaining to Functions:
+
+* FaaS-Admin - allowed to read `lambdas`, manage the whitelist for external domains and manage secrets. This permission is by default set to 'ON' for all admins on the account. It is however possible to limit your admin's access as well. You can create a separate permission group for developers you wish to have part of the admin profile permissions and assign them with the relevant permissions as needed, by creating a custom profile.
+
+* FaaS-Developer - allowed to manage `lambdas`, read whitelisted domains and read encrypted secrets.
+
+* FaaS-Invocation - only allowed to invoke lambdas externally. This permission is granted to a user dedicated to external invocations. When an external system needs to invoke a Functions `lambda`, the external system needs to use an account user with this permission <strong>enabled</strong> to access the LivePerson platform.
```
[the remark this answers is not in the document]
Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:17]: Don't need to go into greater detail. People will learn what they need, under external invocations.

**Review by Templum — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md [hunk 2]
```diff
@@ -8,7 +8,199 @@ permalink: liveperson-functions-permission-system.html
...
+    <tr>
+    <td>FaaS-Developer</td>
+    <td style="text-align: center; vertical-align: middle;" >❌</td>
+    <td style="text-align: center; vertical-align: middle;" >🔒</td>
+    <td style="text-align: center; vertical-align: middle;" >❌</td>
+    <td style="text-align: center; vertical-align: middle;" >✅</td>
+  </tr>
+    <tr>
+    <td>FaaS-Invocation</td>
+    <td style="text-align: center; vertical-align: middle;" >🔒</td>
```
[the remark this answers is not in the document]
Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:54]: Maybe we need to add a s short Legend under the tables

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ There are 3 user permissions pertaining to Functions:
 
 * FaaS-Developer - allowed to manage `lambdas`, read whitelisted domains and read encrypted secrets.
 
-* FaaS-Invocation -provides basic read access on all functions and permission to invoke them. The agent profile has this permission enabled by default.
+* FaaS-Invocation -provides basic read access on all functions and permission to invoke them. The Agent profile has this permission enabled by default.
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:17]:
```suggestion
* FaaS-Invocation - provides basic read access on all functions and permission to invoke them. The Agent profile has this permission enabled by default.
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md [hunk 4]
```diff
@@ -8,7 +8,203 @@ permalink: liveperson-functions-permission-system.html
...
+
+* Click on the users tab at the top of the Conversational Cloud UI and then click profiles.
+
+* Create/Edit a profile and add the relevant FaaS-* permission to it.
+
+![FaaSPermissionSystem](img/faas-permission-system.png)
+
+### Hints
+
+* Be aware that user roles like Agent won't have the option to get FaaS-Developer and FaaS-Admin permission, you would need to use a different user with the proper role.
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:207]:
```suggestion
* Please be aware that roles like Agent don't have the option to get FaaS Developer/Admin permission. You will need to use a different role in that case.
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md [hunk 5]
```diff
@@ -8,7 +8,203 @@ permalink: liveperson-functions-permission-system.html
...
+* CLI users are following the same rules. If you create a user to be used in a CI development it requires the same permissions.
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:208]: Maybe you want to link `CLI` directly to the CLI Page

```suggestion
* The CLI leverages the same permissions, therefore you need to make sure that the user used has at least the FaaS-Developer permission.
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md [hunk 6]
```diff
@@ -8,7 +8,203 @@ permalink: liveperson-functions-permission-system.html
...
+* Use only the minimum permission required. E.g, for external invocation use only a user with FaaS-Invocation permission.
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:209]:
```suggestion
* We suggest obeying the security principle of least privilege and therefore only selecting the minimal set of permissions. E.g. If the user will be only using function (invoking). Then the FaaS-Invocation is only required.
```

**Review by Templum — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md (hunk 4 above)
Comment (eplazaso) [reply pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md:207]: actually is should be a profile Maybe the last sentence could be: _You will need to use another profile with a different role in that case_

**Review by Templum — APPROVED** [review]

**Pull request merged**